File: rust/examples/runFunctions.rs
mod utils;
use std::error::Error;
use utils::get_tls_options;

use aristech_nlp_client::{
    get_client,
    nlp_service::{Function, RunFunctionsRequest},
    run_functions,
};

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    // Load environment variables from .env file
    dotenv::dotenv().ok();

    let host = std::env::var("HOST")?;
    let tls_options = get_tls_options()?;
    let mut client = get_client(host, tls_options).await?;

    let response = run_functions(
        &mut client,
        RunFunctionsRequest {
            input: "hello world".to_string(),
            functions: vec![Function {
                id: "spellcheck-de".to_string(),
                ..Function::default()
            }],
            ..RunFunctionsRequest::default()
        },
    )
    .await?;
    println!("{}", response.output);
    Ok(())
}

File: python/examples/run_functions.py
from aristech_nlp_client import NlpClient, Function, RunFunctionsRequest
from utils import host, auth_token, auth_secret, root_cert, ssl

client = NlpClient(host=host, ssl=ssl, root_cert=root_cert, auth_token=auth_token, auth_secret=auth_secret)
response = client.run_functions(RunFunctionsRequest(
    input="hallo wie geht es dir",
    functions=[
        Function(id="spell")
    ]
))
print(response.output)

client.close()